### ISSUE TITLE
code coverage is very slow with Python 3.12

### ISSUE THREAD

**Issue by berquist** [cclib/cclib#1347]
See https://github.com/cclib/cclib/issues/1305#issuecomment-1902859195.  Let's track everything here.  I'll collect concrete numbers again, but I recall test time went from ~6 minutes to 5 hours on unit/regression tests.  This wasn't just in CI but also on my personal machine outside of Docker.

cc @Andrew-S-Rosen

**Comment by berquist** [comment]
For unit tests with `-m 'not is_method'`:

| version | coverage? | time (s) |
|---------|-----------|----------|
| 11.7    | no        | 30.68    |
| 11.7    | yes       | 52.31    |
| 12.1    | no        | 30.45    |
| 12.1    | yes       | 2400 and counting      |

Tracking issues:
- https://github.com/nedbat/coveragepy/issues/1665
- https://github.com/python/cpython/issues/107674

```
$ coverage --help
Coverage.py, version 7.4.0 with C extension
$ coverage debug sys
-- sys -------------------------------------------------------
               coverage_version: 7.4.0
                coverage_module: /home/eric/.pyenv/versions/miniforge3-4.10.3-10/envs/cclib_312/lib/python3.12/site-packages/coverage/__init__.py
                           core: -none-
                        CTracer: available
```

**Comment by Andrew-S-Rosen** [comment]
This is consistent with my testing as well. When I ran aspects of my suite "manually" without using coverage reports or even pytest, the speed seemed okay.

**Comment by berquist** [comment]
https://github.com/python-pillow/Pillow/pull/7820

**Comment by Andrew-S-Rosen** [comment]
@berquist did you ever figure out how to address this?

**Comment by Andrew-S-Rosen** [comment]
Ah, figured it out.

```
      - name: Set coverage core
        run: |
          echo "COVERAGE_CORE=sysmon" >> $GITHUB_ENV
```

**Comment by berquist** [comment]
The CI setup is done in https://github.com/shivupa/cclib-ci/pull/33 and https://github.com/cclib/cclib/pull/1520.  It's still slower than all other tested Python versions but :shrug: there are a few regression tests that seem particularly bad and I suspect that some actually slow code in cclib is finally exposing itself.  It's also tempting to just turn coverage off for 3.12, but I read that 3.13 isn't an improvement, so this problem will persist.

I haven't added 3.13 to CI yet since Psi4 isn't available with it and our test suite requires dependencies for bridges.  Still thinking about if that should be changed despite it causing a lot of historical pain.

**Comment by berquist** [comment]
Nope, this is going to be unacceptably slow.  I'm going to turn off collecting coverage for 3.12 in CI.
